rcl/src/rcl/node.c

```diff
@@ -72,18 +79,46 @@ rcl_node_init(rcl_node_t * node, const char * name, const rcl_node_options_t * o
     allocator->allocate, "allocate not set", return RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_FOR_NULL_WITH_MSG(
     allocator->deallocate, "deallocate not set", return RCL_RET_INVALID_ARGUMENT);
+  // Make sure the node name is valid before allocating memory.
+  int validation_result = 0;
+  size_t invalid_index = 0;
+  ret = rmw_validate_node_name(name, &validation_result, &invalid_index);
```

Comment:
Since `invalid_index` isn't used it would be good to make the argument optional in the `rmw_validate_node_name` function.